src/x86/avx.rs

```diff
@@ -269,6 +336,383 @@ pub unsafe fn _mm256_sqrt_pd(a: f64x4) -> f64x4 {
+    vcvtpd2ps(a)
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in `a`
+/// to packed 32-bit integers.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vcvtps2dq))]
+pub unsafe fn _mm256_cvtps_epi32(a: f32x8) -> i32x8 {
+    vcvtps2dq(a)
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in `a`
+/// to packed double-precision (64-bit) floating-point elements.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vcvtps2pd))]
+pub unsafe fn _mm256_cvtps_pd(a: f32x4) -> f64x4 {
+    a.as_f64x4()
+}
+
+/// Convert packed double-precision (64-bit) floating-point elements in `a`
+/// to packed 32-bit integers with truncation.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vcvttpd2dq))]
+pub unsafe fn _mm256_cvttpd_epi32(a: f64x4) -> i32x4 {
+    vcvttpd2dq(a)
+}
+
+/// Convert packed double-precision (64-bit) floating-point elements in `a`
+/// to packed 32-bit integers.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vcvtpd2dq))]
+pub unsafe fn _mm256_cvtpd_epi32(a: f64x4) -> i32x4 {
+    vcvtpd2dq(a)
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in `a`
+/// to packed 32-bit integers with truncation.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vcvttps2dq))]
+pub unsafe fn _mm256_cvttps_epi32(a: f32x8) -> i32x8 {
+    vcvttps2dq(a)
+}
+
+/// Extract 128 bits (composed of 4 packed single-precision (32-bit)
+/// floating-point elements) from `a`, selected with `imm8`.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vextractf128))]
+pub unsafe fn _mm256_extractf128_ps(a: f32x8, imm8: i32) -> f32x4 {
+    match imm8 & 1 {
+        0 => simd_shuffle4(a, _mm256_undefined_ps(), [0, 1, 2, 3]),
+        _ => simd_shuffle4(a, _mm256_undefined_ps(), [4, 5, 6, 7]),
+    }
+}
+
+/// Extract 128 bits (composed of 2 packed double-precision (64-bit)
+/// floating-point elements) from `a`, selected with `imm8`.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vextractf128))]
+pub unsafe fn _mm256_extractf128_pd(a: f64x4, imm8: i32) -> f64x2 {
+    match imm8 & 1 {
+        0 => simd_shuffle2(a, _mm256_undefined_pd(), [0, 1]),
+        _ => simd_shuffle2(a, _mm256_undefined_pd(), [2, 3]),
+    }
+}
+
+/// Extract 128 bits (composed of integer data) from `a`, selected with `imm8`.
+#[inline(always)]
+#[target_feature = "+avx"]
+#[cfg_attr(test, assert_instr(vextractf128))]
+pub unsafe fn _mm256_extractf128_si256(a: i64x4, imm8: i32) -> i64x2 {
+    match imm8 & 1 {
+        0 => simd_shuffle2(a, _mm256_undefined_si256(), [0, 1]),
+        _ => simd_shuffle2(a, _mm256_undefined_si256(), [2, 3]),
+    }
+}
+
+/// Extract an 8-bit integer from `a`, selected with `imm8`.
+#[inline(always)]
+#[target_feature = "+avx"]
+pub unsafe fn _mm256_extract_epi8(a: i8x32, imm8: i32) -> i32 {
```

Comment:
Should this and some of the extractions below assert for a particular instruction?

Reply:
It appears that there is [no instrunction specified](https://gist.github.com/anonymous/25d752fda8521d29699a826b980218fc#file-intel-intrinsics-3-3-15-xml-L14777-L14788) for this intrinsic.